packages/typescript-estree/src/convert.ts

```diff
@@ -2365,18 +2367,35 @@ export class Converter {
       }
 
       case SyntaxKind.TypePredicate: {
-        const result = this.createNode<TSESTree.TSTypePredicate>(node, {
-          type: AST_NODE_TYPES.TSTypePredicate,
-          parameterName: this.convertChild(node.parameterName),
-          typeAnnotation: this.convertTypeAnnotation(node.type, node),
+        if (node.assertsModifier) {
+          const result = this.createNode<TSESTree.TSAssertsTypePredicate>(
+            node,
+            {
+              type: AST_NODE_TYPES.TSAssertsTypePredicate,
+              parameterName: this.convertChild(node.parameterName),
+            },
+          );
+          return result;
+        } else if (node.type) {
```

Comment:
Note that it is possible to use **both**.

```ts
function assertFoo(foo: any): asserts foo is Foo
```

https://github.com/microsoft/TypeScript/pull/32695